txnkv/transaction/txn.go

```diff
@@ -395,6 +395,16 @@ func (txn *KVTxn) SetTxnSource(txnSource uint64) {
 	txn.txnSource = txnSource
 }
 
+// SetSessionID sets the session ID of the transaction.
+// If the committer is not initialized yet, the function won't take effect.
+// It is especially useful for pipelined transactions, as its committer is initialized immediately
+// when the transaction is created.
```

Comment:
Should also explain when it is supposed to be executed at tidb-side, is it `OnTxnActiviated`?

Reply:
> is it OnTxnActiviated?

Yes.
But as a lower level module I think it already provides a clear enough interface. It doesn't have to know (or restrict) how it will be used, which should be decided by the user.

Reply:
But the `SessionID` is actually a tidb concept, and it's set in a tricky way using context from the upper layer.

Reply:
Yeah it's a pass-through variable, always annoying to deal with.